// @specodec/typespec-emitter-core/dist/test-utils.js
// Shared mock Type factories for emitter unit testing.
// Import from '@specodec/typespec-emitter-core/test-utils'
export function mkScalar(name) { return { kind: "Scalar", name }; }
export function mkModel(name, props) {
    const properties = new Map();
    if (props)
        for (const [k, v] of Object.entries(props)) {
            if (v && 'kind' in v)
                properties.set(k, { type: v, name: k });
            else if (v && 'type' in v)
                properties.set(k, v);
        }
    return { kind: "Model", name, properties };
}
export function mkEnum(name, members) {
    return { kind: "Enum", name, members: members.map((m, i) => ({ name: m, value: i })) };
}
export function mkArray(elem) {
    return { kind: "Model", name: "Array", indexer: { key: { name: "integer" }, value: elem } };
}
export function mkRecord(val) {
    return { kind: "Model", name: "Record", indexer: { key: { name: "string" }, value: val } };
}
export const ALL_SCALARS = ["string", "boolean", "bytes", "int8", "int16", "int32", "int64", "uint8", "uint16", "uint32", "uint64", "float32", "float64", "float", "decimal", "integer"];
